### PR TITLE
Extracted WebglShader and WebglTexture from WebglGraphicsDevice

### PR DESCRIPTION
**Pull request by mvaligursky**
- Shader functionality related to WebGL has been extracted from WebglGraphicsDevice class into separate WebglShader class.
- Texture functionality related to WebGL has been extracted from WebglGraphicsDevice class into separate WebglTexture class.
- the way textures handle lost context / their tracking by the device is now unified with other resources

### CODE REVIEW

**Review by ellthompson — APPROVED** [review]

File: src/graphics/graphics-device.js [hunk 1]
```diff
@@ -13,6 +13,9 @@ import { Debug } from '../core/debug.js';
 const EVENT_RESIZE = 'resizecanvas';
 
 /** @typedef {import('./render-target.js').RenderTarget} RenderTarget */
+/** @typedef {import('./texture.js').Texture} Texture */
+/** @typedef {import('./render-target.js').RenderTarget} RenderTarget */
```
Comment (willeastcott) [inline src/graphics/graphics-device.js:17]: You have `RenderTarget` twice. Also, I would put these in alphabetical order, personally.

File: src/graphics/graphics-device.js [hunk 2]
```diff
@@ -106,9 +109,15 @@ class GraphicsDevice extends EventHandler {
         this._maxPixelRatio = 1;
 
         // Array of WebGL objects that need to be re-initialized after a context restore event
```
Comment (willeastcott) [inline src/graphics/graphics-device.js:111]: Maybe drop 'WebGL' here?

Comment (mvaligursky) [reply src/graphics/graphics-device.js]: yep done

File: src/graphics/graphics-device.js (hunk 1 above)
Comment (mvaligursky) [reply src/graphics/graphics-device.js]: done, thanks

File: src/graphics/webgl/webgl-shader.js [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+    /**
+     * Compiles an individual shader.
+     *
+     * @param {string} src - The shader source code.
+     * @param {boolean} isVertexShader - True if the shader is a vertex shader, false if it is a
+     * fragment shader.
+     * @returns {WebGLShader} The compiled shader.
+     * @private
+     */
+    compileShaderSource(device, src, isVertexShader) {
```
Comment (willeastcott) [inline src/graphics/webgl/webgl-shader.js:127]: If this is genuinely private, maybe prepend it with an undrescore?

Comment (mvaligursky) [reply src/graphics/webgl/webgl-shader.js]: done

**Review by willeastcott — APPROVED** [review]

**Pull request merged**